### ISSUE TITLE
[java] UselessOverridingMethod false positive when elevating access modifier

### ISSUE THREAD

**Issue by SamCarlberg** [pmd/pmd#911]
**Affects PMD Version:** 

5.6.1, but I've seen old bug reports as far back as 2006: https://sourceforge.net/p/pmd/bugs/423/

**Rule:**

[UselessOverridingMethod](https://pmd.github.io/latest/pmd_rules_java_design.html#uselessoverridingmethod)

**Description:**  

Overriding a method in a superclass, but only elevating the access modifier (eg `protected` -> `public`), will be reported as a useless override, even though the method in the subclass has a different access modifier.

**Code Sample demonstrating the issue:**

```java
class TestA {
  protected void hello() {
    System.out.println("Hello, world!");
  }
}

class TestB extends TestA {
  @Override
  public void hello() { // Elevate the access level - PMD reports this as useless
    super.hello();
  }
}

```

**Running PMD through:**

Gradle

**Comment by astellingwerf** [comment]
The issue is not limited to straight forward visibility elevation, but also is seen when a `protected` method is overridden to be `protected` in another class in another package.
